WordPress/Classes/ViewRelated/Menus/Views/MenuItemEditingHeaderView.m

```diff
@@ -0,0 +1,212 @@
+#import "MenuItemEditingHeaderView.h"
+#import "WPStyleGuide.h"
+#import "MenuItem.h"
+#import "MenuItem+ViewDesign.h"
+
+@interface MenuItemEditingHeaderView () <UITextFieldDelegate>
+
+@property (nonatomic, strong, readonly) UIStackView *stackView;
+@property (nonatomic, strong, readonly) NSLayoutConstraint *stackViewTopConstraint;
+@property (nonatomic, strong, readonly) UIView *textFieldContainerView;
+@property (nonatomic, strong, readonly) UIImageView *iconView;
+
+@end
+
+@implementation MenuItemEditingHeaderView
+
+- (void)dealloc
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)awakeFromNib
+{
+    [super awakeFromNib];
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(deviceOrientationDidChangeNotification) name:UIDeviceOrientationDidChangeNotification object:nil];
+    
+    self.translatesAutoresizingMaskIntoConstraints = NO;
+    self.backgroundColor = [UIColor clearColor];
+
+    [self setupStackView];
+    [self setupIconView];
+    [self setupTextField];
```

Comment:
Since this is being loaded from the storyboard, why not define and setup the stackview, iconview and textfield in the storyboard itself?  Was there a particular reason to do it all via code?